lib/Auth/Process/ΕvaluateΕntitlement.php

```diff
@@ -0,0 +1,175 @@
+            }
+            $this->attributeName = $config['attributeName'];
+        }
+
+        if (array_key_exists('entitlement', $config)) {
+            if (!is_array($config['entitlement'])) {
+                SimpleSAML_Logger::error("[ΕvaluateΕntitlement] Configuration error: 'entitlement' not a string literal");
+                throw new Exception("ΕvaluateΕntitlement configuration error: 'entitlement' not a string literal");
+            }
+            $this->entitlement = $config['entitlement'];
+        }
+
+        if (array_key_exists('idpBlacklist', $config)) {
+            if (!is_array($config['idpBlacklist'])) {
+                SimpleSAML_Logger::error("[ΕvaluateΕntitlement] Configuration error: 'idpBlacklist' not a string literal");
+                throw new Exception("ΕvaluateΕntitlement configuration error: 'idpBlacklist' not a string literal");
+            }
+            $this->idpBlacklist = $config['idpBlacklist'];
+        }
+
+        if (array_key_exists('idpWhitelist', $config)) {
+            if (!is_array($config['idpWhitelist'])) {
+                SimpleSAML_Logger::error("[ΕvaluateΕntitlement] Configuration error: 'idpWhitelist' not a string literal");
+                throw new Exception("ΕvaluateΕntitlement configuration error: 'idpWhitelist' not a string literal");
+            }
+            $this->idpWhitelist = $config['idpWhitelist'];
+        }
+
+        if (array_key_exists('entityAttributeWhitelist', $config)) {
+            if (!is_array($config['entityAttributeWhitelist'])) {
+                SimpleSAML_Logger::error("[ΕvaluateΕntitlement] Configuration error: 'entityAttributeWhitelist' not a string literal");
+                throw new Exception("ΕvaluateΕntitlement configuration error: 'entityAttributeWhitelist' not a string literal");
+            }
+            $this->entityAttributeWhitelist = $config['entityAttributeWhitelist'];
+        }
+
+        if (array_key_exists('entitlementWhitelist', $config)) {
+            if (!is_array($config['entitlementWhitelist'])) {
+                SimpleSAML_Logger::error("[ΕvaluateΕntitlement] Configuration error: 'entitlementWhitelist' not a string literal");
+                throw new Exception("ΕvaluateΕntitlement configuration error: 'entitlementWhitelist' not a string literal");
+            }
+            $this->entitlementWhitelist = $config['entitlementWhitelist'];
+        }
+    }
+
+
+    /**
+     *
+     * @param array &$state The current SP state
+     */
+    public function process(&$state)
+    {
+        assert('is_array($state)');
+
+        if (!$this->isQualified($state)) {
+            return;
+        }
+        if (empty($state['Attributes'][$this->attributeName])) {
+            $state['Attributes'][$this->attributeName] = array();
+        }
+        $state['Attributes'][$this->attributeName] = array_merge($state['Attributes'][$this->attributeName], $this->entitlement);
```

Comment:
`$this->entitlement` --> `$this-->capability`

Reply:
Add debug: `"[AddCapability] Adding capability " . var_export($this-->capability, true)"`